delivery_price_rule_volumetric_weight/models/delivery_carrier.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.safe_eval import safe_eval
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    def _get_price_available(self, order):
+        self.ensure_one()
+        self = self.sudo()
+        order = order.sudo()
+        total = weight = volumetric_weight = volume = quantity = 0
+        total_delivery = 0.0
+        for line in order.order_line.filtered(
+            lambda x: x.state != "cancel" and x.product_id
+        ):
+            if line.is_delivery:
+                total_delivery += line.price_total
+                continue
+            qty = line.product_uom._compute_quantity(
+                line.product_uom_qty, line.product_id.uom_id
+            )
+            weight += (line.product_id.weight or 0.0) * qty
+            volumetric_weight += (line.product_id.volumetric_weight or 0.0) * qty
+            volume += (line.product_id.volume or 0.0) * qty
+            quantity += qty
+        total = (order.amount_total or 0.0) - total_delivery
+        total = order.currency_id._convert(
+            total,
+            order.company_id.currency_id,
+            order.company_id,
+            order.date_order or fields.Date.today(),
+        )
+        _self = self.with_context(volumetric_weight=volumetric_weight)
+        return _self._get_price_from_picking(total, weight, volume, quantity)
+
+    def _get_price_from_picking(self, total, weight, volume, quantity):
+        price = 0.0
+        criteria_found = False
+        price_dict = {
```

Comment:
You can try to propose to Odoo a hook that return the `price_dict` dictionary for not having to override whole method.

Reply:
https://github.com/odoo/odoo/pull/74529